libraries/chain/asset.cpp

```diff
@@ -80,13 +77,42 @@ asset asset::from_string(const string& from)
       return result;
    }
    FC_CAPTURE_LOG_AND_RETHROW( (from) )
+   */
+   try { 
+      string s = fc::trim( from );
+      auto space_pos = s.find( " " );
+      auto dot_pos = s.find( "." );
+      
+      asset result;
+      
+      auto intpart = s.substr( 0, dot_pos );
+      result.amount = fc::to_int64(intpart);
+      string symbol_part;
+      if (dot_pos != string::npos && space_pos != string::npos)
+         {
+            symbol_part = eosio::chain::to_string(space_pos - dot_pos - 1);
+            symbol_part += ',';
+            symbol_part += s.substr(space_pos + 1);
+         }
+      result.sym = symbol::from_string(symbol_part);
+      if( dot_pos != string::npos )
+         {
+            auto fractpart = "1" + s.substr( dot_pos + 1, space_pos - dot_pos - 1 );
+            
+            result.amount *= int64_t(result.precision());
+            result.amount += int64_t(fc::to_int64(fractpart));
+            result.amount -= int64_t(result.precision());
+         }
```

Comment:
Code formatting for `if` here does not follow the rest of the codebase.

Reply:
Changed indentation.